ui/src/alerting/actions/checks.ts

```diff
@@ -155,7 +155,12 @@ export const saveCheckFromTimeMachine = () => async (
     ...check,
     query: draftQueries[0],
     orgID,
-    labels: check.labels.map(l => l.id),
+    labels:
+      (check &&
+        check.labels &&
+        Array.isArray(check.labels) &&
+        check.labels.map(l => l.id)) ||
+      [],
```

Comment:
if this is too nitpicky, tell me to stfu and i'll gladly do it! 😃 

from what i've seen in the codebase, i think the more idiomatic way to do this is something like:

```js
import {get} from 'lodash'

//...

const labels = get(check, 'labels', [])

const data = {
  ...check,
  name: clonedName,
  labels: labels.map(l => l.id),
} as PostCheck
```
lodash's [`get`](https://lodash.com/docs/4.17.15#get) will check the first argument (`check`) for the second argument (`'labels'`). it'll do the falsey checks you're doing as it goes down the property list. if it can't find the property you're looking for, it returns the third property - an empty array.

(it's not a huge deal for a single property object, but it's super valuable when there are deeply nested properties, like [here,](https://github.com/influxdata/influxdb/blob/master/ui/src/dashboards/selectors/index.ts#L12)).

using `get` like above, it'll guard against falsey values (`null`, `undefined`, `false`, `''`, etc), and includes the `) || []` that ensures labels stays an array. but it won't guard against truthy values, so if `labels` is anything but an array, like a string or an object, this will fail. but if you search `src/client/generatedRoutes` for `export type Labels`, you'll see it's typed as an array `Label[]`, so i think this check is safe.

the reason i bring it up is i think it's probably a little bit safer and less likely to get tripped up on an edge case, and because we use this pattern (using `get`) in quite a few places i've seen in the code base, it's fairly well understood.

Reply:
I ❤️ `get` too